components/src/core/ChannelValue/ChannelValue.tsx

```diff
@@ -87,6 +105,19 @@ const ChannelValueRender: React.ForwardRefRenderFunction<unknown, ChannelValuePr
         ...otherSpanProps
     } = props;
     const defaultClasses = useStyles(props);
+    const prefixUnitWhitelist = ['$'];
+    const suffixUnitWhitelist = ['%', '℉', '°F', '℃', '°C', '°'];
+    const applyPrefix = useCallback((): boolean => {
+        if ((!prefixUnitWhitelist.includes(units) && unitSpace !== 'hide') || unitSpace === 'show') {
+            return true;
+        }
+    }, [prefix, units, unitSpace]);
+
+    const applySuffix = useCallback((): boolean => {
```

Comment:
```
const applyPrefix = useCallback((): boolean => 
    (unitSpace !== 'hide' && (unitSpace === 'show' || !prefixUnitWhitelist.includes(units))
}, [prefix, units, unitSpace]);
```

Maybe a little simpler formatting above.

Also does unitSpace need to be in this deps list?

Can we update the whiteList variable to be a little more specific; I'm having a really hard time understand what it means for a variable to be a part of this list.  Perhaps something like `allowSpaceList`?

Reply:
Please check https://github.com/brightlayer-ui/react-component-library/pull/362/commits/09589f583a1fb48c662c2e0af95f7e886ab940fe#diff-9b02b302c55684c472c77ad7327b46f3aa673c8fbda241554b4a144edd0eeacdR39